Exception while using snapshot agent mode

```
2022-11-15  23:09:46.932+01:00     ERROR     [Equinox Log Thread - Equinox Container: c5707ab5-3d3f-4d33-ae94-d6d5f6a46a07]  [LogService.org.eclipse.e4.ui.workbench]  Unable to create class 'com.osgifx.console.ui.events.handler.EventReceiveMenuContributionHandler' from bundle '103' (org.eclipse.e4.core.di.InjectionException: org.apache.commons.lang.NotImplementedException: Snapshot supervisor doesn't require this functionality)
org.eclipse.e4.core.di.InjectionException: org.apache.commons.lang.NotImplementedException: Snapshot supervisor doesn't require this functionality
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:68)
	at org.eclipse.e4.core.internal.di.InjectorImpl.processAnnotated(InjectorImpl.java:995)
	at org.eclipse.e4.core.internal.di.InjectorImpl.internalInject(InjectorImpl.java:140)
	at org.eclipse.e4.core.internal.di.InjectorImpl.internalMake(InjectorImpl.java:403)
	at org.eclipse.e4.core.internal.di.InjectorImpl.make(InjectorImpl.java:330)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.make(ContextInjectionFactory.java:202)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.createFromBundle(ReflectionContributionFactory.java:91)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.doCreate(ReflectionContributionFactory.java:60)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.create(ReflectionContributionFactory.java:42)
	at org.eclipse.fx.ui.workbench.renderers.base.BaseItemRenderer.getContributionObject(BaseItemRenderer.java:236)
	at org.eclipse.fx.ui.workbench.renderers.base.BaseItemRenderer.canExecute(BaseItemRenderer.java:138)
	at org.eclipse.fx.ui.workbench.renderers.base.BaseMenuItemRenderer$2.run(BaseMenuItemRenderer.java:114)
	at org.eclipse.fx.ui.controls.internal.FXThreadSynchronizeImpl.syncExec(FXThreadSynchronizeImpl.java:109)
	at org.eclipse.fx.ui.workbench.fx.internal.UISynchronizeImpl.syncExec(UISynchronizeImpl.java:51)
	at org.eclipse.fx.ui.workbench.renderers.base.BaseMenuItemRenderer.checkEnablement(BaseMenuItemRenderer.java:107)
	at org.eclipse.fx.ui.workbench.renderers.base.BaseMenuItemRenderer.checkEnablement(BaseMenuItemRenderer.java:1)
	at org.eclipse.fx.ui.workbench.renderers.base.BaseMenuRenderer.handleShowing(BaseMenuRenderer.java:170)
	at org.eclipse.fx.ui.workbench.renderers.base.BaseMenuRenderer.lambda$0(BaseMenuRenderer.java:80)
	at org.eclipse.fx.ui.workbench.renderers.fx.DefMenuRenderer$MenuImpl$1.handle(DefMenuRenderer.java:75)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.controls/javafx.scene.control.Menu.setShowing(Menu.java:205)
	at javafx.controls/javafx.scene.control.Menu.show(Menu.java:412)
	at javafx.controls/javafx.scene.control.skin.MenuBarSkin.showMenu(MenuBarSkin.java:471)
	at javafx.controls/javafx.scene.control.skin.MenuBarSkin.showMenu(MenuBarSkin.java:450)
	at javafx.controls/javafx.scene.control.skin.MenuBarSkin.lambda$rebuildUI$29(MenuBarSkin.java:1013)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.base/com.sun.javafx.event.EventQueue.fire(EventQueue.java:48)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.handleEnterExit(Scene.java:3811)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.process(Scene.java:3882)
	at javafx.graphics/javafx.scene.Scene.processMouseEvent(Scene.java:1878)
	at javafx.graphics/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2623)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:411)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:301)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:450)
	at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:449)
	at javafx.graphics/com.sun.glass.ui.View.handleMouseEvent(View.java:557)
	at javafx.graphics/com.sun.glass.ui.View.notifyMouse(View.java:943)
	at javafx.graphics/com.sun.glass.ui.mac.MacView.notifyMouse(MacView.java:127)
Caused by: org.apache.commons.lang.NotImplementedException: Snapshot supervisor doesn't require this functionality
	at com.osgifx.console.supervisor.snapshot.SnapshotSupervisor.addOSGiEventListener(SnapshotSupervisor.java:82)
	at com.osgifx.console.ui.events.handler.EventReceiveMenuContributionHandler.init(EventReceiveMenuContributionHandler.java:104)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	... 60 common frames omitted
```

--- a/com.osgifx.console.supervisor/src/main/java/com/osgifx/console/supervisor/snapshot/SnapshotSupervisor.java
+++ b/com.osgifx.console.supervisor/src/main/java/com/osgifx/console/supervisor/snapshot/SnapshotSupervisor.java
@@ -20,8 +20,10 @@ import static org.osgi.service.condition.Condition.CONDITION_ID;
 
 import java.io.IOException;
 
-import org.apache.commons.lang.NotImplementedException;
+import org.eclipse.fx.core.log.FluentLogger;
+import org.eclipse.fx.core.log.LoggerFactory;
 import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Deactivate;
 import org.osgi.service.component.annotations.Reference;
@@ -38,13 +40,22 @@ import com.osgifx.console.supervisor.Supervisor;
 @SatisfyingConditionTarget("(" + CONDITION_ID + "=" + CONDITION_ID_VALUE + ")")
 public final class SnapshotSupervisor implements Supervisor {
 
-    public static final String CONDITION_ID_VALUE = "snapshot-agent";
+    public static final String  CONDITION_ID_VALUE  = "snapshot-agent";
+    private static final String NOT_IMPLEMENTED_LOG = "Snapshot supervisor doesn't require this functionality";
 
     @Reference
     private SnapshotAgent agent;
+    @Reference
+    private LoggerFactory factory;
+    private FluentLogger  logger;
 
     @Reference
     private ConfigurationAdmin configAdmin;
+
+    @Activate
+    void activate() {
+        logger = FluentLogger.of(factory.createLogger(getClass().getName()));
+    }
 
     @Deactivate
     void deactivate() throws IOException {
@@ -64,37 +75,37 @@ public final class SnapshotSupervisor implements Supervisor {
 
     @Override
     public void connect(final String host, final int port, final int timeout) throws Exception {
-        throw new NotImplementedException("Snapshot supervisor doesn't require this functionality");
+        logger.atInfo().log(NOT_IMPLEMENTED_LOG);
     }
 
     @Override
     public void onOSGiEvent(final XEventDTO event) {
-        throw new NotImplementedException("Snapshot supervisor doesn't require this functionality");
+        logger.atInfo().log(NOT_IMPLEMENTED_LOG);
     }
 
     @Override
     public void logged(final XLogEntryDTO event) {
-        throw new NotImplementedException("Snapshot supervisor doesn't require this functionality");
+        logger.atInfo().log(NOT_IMPLEMENTED_LOG);
     }
 
     @Override
     public void addOSGiEventListener(final EventListener eventListener) {
-        throw new NotImplementedException("Snapshot supervisor doesn't require this functionality");
+        logger.atInfo().log(NOT_IMPLEMENTED_LOG);
     }
 
     @Override
     public void removeOSGiEventListener(final EventListener eventListener) {
-        throw new NotImplementedException("Snapshot supervisor doesn't require this functionality");
+        logger.atInfo().log(NOT_IMPLEMENTED_LOG);
     }
 
     @Override
     public void addOSGiLogListener(final LogEntryListener logEntryListener) {
-        throw new NotImplementedException("Snapshot supervisor doesn't require this functionality");
+        logger.atInfo().log(NOT_IMPLEMENTED_LOG);
     }
 
     @Override
     public void removeOSGiLogListener(final LogEntryListener logEntryListener) {
-        throw new NotImplementedException("Snapshot supervisor doesn't require this functionality");
+        logger.atInfo().log(NOT_IMPLEMENTED_LOG);
     }
 
     @Override

